### PR TITLE
Bug 1527437 [Cards][Lists] - Align spocs items to bottom of container

### PR DESCRIPTION
**Pull request by punamdahiya**

### CODE REVIEW

**Comment by punamdahiya** [comment]
@Mardak @gvn please review. Thanks!

**Comment by Mardak** [comment]
This should have already been fixed for cards with #4737. @ScottDowne did something change to remove that functionality but we want it back again?

**Comment by Mardak** [comment]
Ah, bisecting the regression points to #4774 which probably has the same underlying issue as the fixes for #4782 and #4785. Although the latter one #4785 changes the parent order again and didn't seem to fix…

**Comment by Mardak** [comment]
We probably don't even need to do this as a `React.Fragment` if we make users of `<original stuff><SpocIntersectionObserver />` instead of `<SpocIntersectionObserver><original stuff></>` although that would touch a lot of lines and probably should be avoided for uplifts. @k88hudson ?

**Comment by Mardak** [comment]
Oh… thecount pointed out that @ncloudioj is removing `SpocIntersectionObserver` with #4743 so this PR probably should look for fixes that don't depend on stuff in there.

**Comment by punamdahiya** [comment]
> Oh… thecount pointed out that @ncloudioj is removing `SpocIntersectionObserver` with #4743 so this PR probably should look for fixes that don't depend on stuff in there.

I agree if we are planning to remove SpocIntersectionObserver, we should not have a className dependency in there. I believe  rendering as below is a better approach. Will update PR. Thanks!
```
  <li className="ds-list-item">
    <original stuff>
    <SpocIntersectionObserver />
  </li>
```

**Comment by gvn** [comment]
@punamdahiya Can you add QA steps?

**Comment by punamdahiya** [comment]
@Mardak Updated PR with review feedback, thanks!

@gvn To view spoc item in list and hero components use layout_endpoint below "layout_endpoint":"https://gist.githubusercontent.com/punamdahiya/4852948f6b24e1a82f4cc2f107ddc234/raw/7e54aa261eeeb718b94a55965b1da28348df93dc/feed.json"

To verify Card layout switch endpoint from basic to dev-test-all

Thanks!

File: content-src/components/DiscoveryStreamComponents/List/_List.scss [hunk 1]
```diff
@@ -162,6 +162,7 @@ $item-line-height: 20;
 
     display: flex;
     justify-content: space-between;
+    height: 100%;
```
Comment (gvn) [inline content-src/components/DiscoveryStreamComponents/List/_List.scss]: Is this necessary?

Comment (punamdahiya) [reply content-src/components/DiscoveryStreamComponents/List/_List.scss]: yes, without this link doesn't take full height from parent and fails to bottom align source.

**Pull request merged**